most.js

```diff
@@ -1,42 +1,28 @@
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
+const most          = require('most');
 
-const most = require('most');
-const R = require('ramda');
-const S = require('sanctuary');
+const exit0         = require('./common/exit0');
+const exit1         = require('./common/exit1');
+const join          = require('./common/join');
+const readFile      = require('./common/read-file-promise');
+const S             = require('./common/sanctuary');
 
 
-//    join :: String -> String -> String
-const join = R.curryN(2, path.join);
-
-//    readFile :: String -> String -> Promise Error String
-const readFile = R.curry((encoding, filename) =>
-  new Promise((resolve, reject) =>
-    fs.readFile(filename, { encoding: encoding }, (err, data) =>
-      err != null ? reject(err) : resolve(data)
-    )));
-
-//    concatFiles :: String -> Promise Error String
-const concatFiles = dir =>
-  R.pipe(R.pipe(readFile('utf8'), most.fromPromise),
-         R.map(S.lines),
-         R.map(R.map(join(dir))),
-         R.chain(R.pipe(R.map(readFile('utf8')), most.from)),
-         stream => stream.await(),
-         R.reduce(R.concat, '')
-  )(join(dir, 'index.txt'));
+//    concatFiles :: (String -> String) -> Promise Error String
+const concatFiles = path =>
+  most.fromPromise(readFile(path('index.txt')))
+  .map(S.lines)
+  .map(S.map(path))
+  .map(S.map(readFile))
+  .map(Promise.all.bind(Promise))
+  .awaitPromises()
```

Comment:
This seems much better (although the reducing function should be `S.concat` rather than `S.joinWith('')`). When I try it in one of the failure cases, though, I see:

  - `UnhandledPromiseRejectionWarning: Unhandled promise rejection`
  - `DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`